### PR TITLE
[i228] Support auto-translation feature in XML

### PR DESCRIPTION
**Pull request by kanat**
### 🎯 Goal

Closes:
- https://github.com/GetStream/android-internal-board/issues/228

Relates to:
- https://github.com/GetStream/stream-chat-android/pull/5028

### 🎨 UI Changes

| Before | After |
| --- | --- |
| XML_EN | XML_FR |

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

**Comment by kanat** [comment]
I need to add docs, but implementation-wise is ready to be reviewed.

**Comment by kanat** [comment]
Docs have beed added 🔥

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/utils/extensions/Message.kt [hunk 1]
```diff
@@ -123,3 +125,25 @@ private fun getAttachmentPrefix(attachment: Attachment): String? =
...
         else -> null
     }
+
+internal fun Message.getTranslatedText(currentUser: User?): String {
+    return when (ChatUI.autoTranslationEnabled) {
+        true -> currentUser?.language?.let { userLanguage ->
+            getTranslation(userLanguage).ifEmpty { text }
+        } ?: text
+        else -> text
+    }
```
Comment (JcMinarro) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/utils/extensions/Message.kt:135]:
```suggestion
    return getTranslatedText { currentUser }
```

Comment (kanat) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/utils/extensions/Message.kt:135]: Nice catch @JcMinarro 👍 
Thank you ❤️

Comment (kanat) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/utils/extensions/Message.kt]: I've also made the the following function as `inline` one to reduce the overhead 👍 
```kotlin
internal inline fun Message.getTranslatedText(getCurrentUser: () -> User?): String
```

**Review by JcMinarro — APPROVED** [review]

**Pull request merged**